pydantic/_internal/_generate_schema.py

```diff
@@ -742,7 +742,7 @@ def _generate_parameter_schema(
         else:
             field = FieldInfo.from_annotated_attribute(annotation, default)
         assert field.annotation is not None, 'field.annotation should not be None when generating a schema'
-        source_type, annotations = annotation, field.metadata
+        source_type, annotations = field.annotation, field.metadata
```

Comment:
even if there are issues merging the rest of this, we should probably make this fix either way.